src/main/java/org/apache/sysds/runtime/controlprogram/federated/FederationMap.java

```diff
@@ -226,11 +228,27 @@ public FederatedRequest broadcast(ScalarObject scalar) {
 		// multi-threaded block slicing and federation request creation
 		FederatedRequest[] ret = new FederatedRequest[ix.length];
 		Arrays.setAll(ret,
-			i -> new FederatedRequest(RequestType.PUT_VAR, lineageTrace, id,
-				cb.slice(ix[i][0], ix[i][1], ix[i][2], ix[i][3], isFrame ? new FrameBlock() : new MatrixBlock())));
+			i -> sliceBroadcastBlock(ix[i], id, cb, lineageItem, isFrame));
 		return ret;
 	}
 
+	private FederatedRequest sliceBroadcastBlock(int[] ix, long id, CacheBlock cb, LineageItem objLi, boolean isFrame) {
+		LineageItem li = null;
+		if(objLi != null) {
+			final String scalarCPOSuffix = InstructionUtils.concatOperands(
+				DataType.SCALAR.name(), ValueType.INT64.name(), String.valueOf(true));
+			// manually create a lineage item for indexing to complete the lineage trace for slicing
+			li = new LineageItem(RightIndex.OPCODE, new LineageItem[]{objLi,
+				new LineageItem(InstructionUtils.concatOperandParts(String.valueOf(ix[0] + 1), scalarCPOSuffix)),
+				new LineageItem(InstructionUtils.concatOperandParts(String.valueOf(ix[1] + 1), scalarCPOSuffix)),
+				new LineageItem(InstructionUtils.concatOperandParts(String.valueOf(ix[2] + 1), scalarCPOSuffix)),
+				new LineageItem(InstructionUtils.concatOperandParts(String.valueOf(ix[3] + 1), scalarCPOSuffix))});
```

Comment:
Please follow how we create lineage items in ParameterizedBuiltinCPInstruction. You simply need to create a CPOperand for each input and pass those to the getLineage function.

Reply:
Thank you for the advice @phaniarnab. I modified it to create CPOperand for each index and derive the LineageItem from the CPOperands. However, using the getLineage() method to obtain the LineageItems is not that easy, as we would need the ExecutionContext for that.

Reply:
You probably need to fix some javadoc warnings @ywcb00. Please have a look at why the tests failed.